hyperspy/tests/signals/test_ragged_signal.py

```diff
@@ -89,6 +89,19 @@ def test_slicing(self):
         with pytest.raises(RuntimeError):
             s.isig[0]
 
+    def test_ragged_0D_map(self):
+        def return_value(val):
+            return val
+
+        zeroDragged = self.s.inav[0, 0]
+
+        new_signal = zeroDragged.map(return_value, inplace=False)
+        if self.s._lazy:
+            new_signal.compute()
+            assert self.s.data[0, 0].compute().shape == new_signal.data[0].shape
+        else:
+            self.s.data[0, 0].shape == new_signal.data[0].shape
```

Comment:
Should it be like this?
```suggestion
        if self.s._lazy:
            new_signal.compute()
            zeroDragged.compute()
        assert zeroDragged.data.shape == new_signal.data.shape
        np.testing.assert_allclose(zeroDragged.data[0], new_signal.data[0])
```

Reply:
So I think the problem is actually slicing the lazy dataset which explains the different behavior for lazy/ non-lazy signals. 

Slicing the dataset causes numpy to try to cast to a non object so somewhere in the task graph is a casting step which needs to be suppressed or reversed

Reply:
I am trying to figure out what should be the desired behaviour first: with the suggested changes above, both cases (lazy and non-lazy) fail but with a different failures. Without the changes it only fails for lazy.

I didn't look any further than that! 😉

Reply:
Sorry I didn't do a good job at describing the problem(s).  

The first is that when you have a 0-D Ragged signal and try to use the map function it will coerce the data into a lazy signal but uses the navigation_axes which is `( )` for a ragged signal. This causes a problem when the signal is cast to a lazy signal. 

More Minimal example:
```
import numpy as np
import hyperspy.api as hs
arr = np.empty((1), dtype=object)
arr[0] = np.array([1,2])
s = hs.signals.BaseSignal(arr, ragged=True)
s_lazy = s.as_lazy()
s_lazy.rechunk() # raises error
```

The second bug is related to a bug (or more likely undefined behavior) upstream in `dask` where numpy is casting from ragged --> not ragged.  This causes issues when you slice and then the object dtype is lost.  

```
import numpy as np
import dask.array as da
arr = np.empty((2), dtype=object)
arr[0] = np.array([1,2])
arr[1] = np.array([1,2,2])
d_arr = da.from_array(arr, chunks=(1,1))
first_ind = d_arr[0]
first_ind =first_ind.compute()
first_ind.dtype #dtype('int64')
first_ind.shape #(2,)
```